bindings/python/src/session.cpp

```diff
@@ -248,8 +248,16 @@ namespace
             }
             else if(key == "info_hash")
             {
-                p.info_hash = sha1_hash(
-                bytes(extract<bytes>(value)).arr.data());
+                if (boost::python::len(value) == 40)
+                {
+                    p.info_hash = info_hash_t(sha1_hash(
+                            bytes(extract<bytes>(value)).arr.data()));
+                }
+                else
+                {
+                    p.info_hash = info_hash_t(sha256_hash(
```

Comment:
maybe there should be a runtime check here, that the buffer is of the right size here too.